docs/source/usage_guides/gradient_accumulation.md

```diff
@@ -138,4 +143,77 @@ for batch in training_dataloader:
         optimizer.zero_grad()
 ```
 
+<Tip warning={true}>
+  It's important that you enter the `with accelerator.accumulate(model)` context manager before you perform the forward and backward pass of your model. In other words, do **not** do this:
+  
+  ```diff
+  -with accelerator.accumulate(model):
+  -    for batch in training_dataloader:
+          # ...
+  ```
+</Tip>
+
+
 To learn more about what magic this wraps around, read the [Gradient Synchronization concept guide](../concept_guides/gradient_synchronization)
+
+
+## Self-contained example
+
+Here is a self-contained example that you can run to see gradient accumulation in action with 🤗 Accelerate:
+
+```python
+import torch
+from torch.utils.data import TensorDataset, DataLoader
+import copy
+from accelerate import Accelerator
+
+# seed
+torch.manual_seed(0)
+
+# define toy inputs and labels
+x = torch.tensor([1., 2., 3., 4., 5., 6., 7., 8.])
+y = torch.tensor([2., 4., 6., 8., 10., 12., 14., 16.])
+gradient_accumulation_steps = 4
+batch_size = int(len(x) // gradient_accumulation_steps)
```

Comment:
Not sure the cast to `int` is needed here since `//` will always return an integer